src/main/java/com/github/princesslana/eriscasper/ErisCasper.java

```diff
@@ -60,13 +60,21 @@ private ErisCasper(BotToken token, Optional<ShardPayload> shard) {
   }
 
   public void run(Bot bot) {
-    Observable<Event> events = getEvents();
-
-    RepositoryManager rm = RepositoryManager.create(events);
-
-    bot.apply(new BotContext(events, routes, rm))
-        .doOnError(t -> LOG.warn("Exception thrown by Bot", t))
-        .blockingAwait();
+    try {
+      getEvents()
+          .compose(
+              evts ->
+                  bot.apply(new BotContext(evts, routes, RepositoryManager.create(evts)))
+                      .toObservable())
+          .doOnError(t -> LOG.warn("Exception thrown by Bot", t))
+          .blockingSubscribe();
+    } finally {
+      Schedulers.shutdown();
+      httpClient.dispatcher().executorService().shutdown();
+      httpClient.connectionPool().evictAll();
+    }
+
+    LOG.trace("ErisCasper.run done.");
```

Comment:
should we change this to a method reference? E.g `Log.trace("ErisCasper#run(Bot) done.");`